pragma-oracle/src/tests/test_oracle.cairo

```diff
@@ -1371,3 +1380,94 @@ fn test_update_pair() {
     assert(pair.quote_currency_id == 111, 'wrong recorded pair');
     assert(pair.base_currency_id == 12345, 'wrong recorded pair');
 }
+
+
+#[test]
+#[available_gas(20000000000000)]
+fn test_register_tokenized_vault() {
+    let (publisher_registry, oracle) = setup();
+    let token: felt252 = 'xStrk';
+    let token_address: ContractAddress = contract_address_const::<0x12345566>();
+    let admin = contract_address_const::<0x123456789>();
+    set_contract_address(admin);
+    oracle.register_tokenized_vault(token, token_address);
+    assert(oracle.get_tokenized_vaults(token) == token_address, 'Failed to register token');
+}
+
+#[test]
+#[available_gas(20000000000000)]
+fn test_delete_tokenized_vault() {
+    let (publisher_registry, oracle) = setup();
+    let token: felt252 = 'xStrk';
+    let token_address: ContractAddress = contract_address_const::<0x12345566>();
+    let admin = contract_address_const::<0x123456789>();
+    set_contract_address(admin);
+    oracle.register_tokenized_vault(token, token_address);
+    assert(oracle.get_tokenized_vaults(token) == token_address, 'Failed to register token');
+    oracle.delete_tokenized_vault(token);
+    assert(
+        oracle.get_tokenized_vaults(token) == contract_address_const::<0>(),
+        'Failed to delete token'
+    );
+}
+
+
+#[test]
+#[available_gas(20000000000000)]
+#[should_panic(expected: ('Admin: unauthorized', 'ENTRYPOINT_FAILED'))]
+fn test_register_tokenized_vault_panics_if_not_owner() {
+    let (publisher_registry, oracle) = setup();
+    let token: felt252 = 'xStrk';
+    let token_address: ContractAddress = contract_address_const::<0x12345566>();
+    set_contract_address(contract_address_const::<0x12>());
+    oracle.register_tokenized_vault(token, token_address);
+}
+
+
+#[test]
+#[available_gas(20000000000000)]
+fn test_get_conversion_rate_price() {
+    let now = 100000;
+    let (publisher_registry, oracle) = setup();
+    let admin = contract_address_const::<0x123456789>();
+    set_contract_address(admin);
+    oracle
+        .add_currency(
+            Currency {
+                id: 'STRK',
+                decimals: 8,
+                is_abstract_currency: false,
+                starknet_address: 0.try_into().unwrap(),
+                ethereum_address: 0.try_into().unwrap(),
+            }
+        );
+    oracle
+        .add_currency(
+            Currency {
+                id: 'xSTRK',
+                decimals: 8,
```

Comment:
```suggestion
                decimals: 18,
```

Reply:
Didn't we set the onchain decimals at 8 ?